File: 06 - Exam Analysis/Vcl/1StClass4000/VCL6/demos/imager/TilerUnit.pas
unit TilerUnit;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  fcOutlookList, fcLabel, ExtCtrls, fcClearPanel
  {$ifndef ver100}, ImgList{$endif}, Buttons, fcImager, StdCtrls, fcButton,
  fcImgBtn, fcShapeBtn, fcButtonGroup, fcOutlookBar;

type
  TTilerForm = class(TForm)
    ImageList1: TImageList;
    fcLabel1: TfcLabel;
    fcImager1: TfcImager;
    fcOutlookBar1: TfcOutlookBar;
    fcOutlookBar1fcShapeBtn1: TfcShapeBtn;
    fcControlBar1OutlookList2: TfcOutlookList;
    procedure fcControlBar1OutlookList1ItemChange(
      OutlookList: TfcCustomOutlookList; Item: TfcOutlookListItem);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  TilerForm: TTilerForm;

implementation

{$R *.DFM}


procedure TTilerForm.fcControlBar1OutlookList1ItemChange(
  OutlookList: TfcCustomOutlookList; Item: TfcOutlookListItem);
var ABitmap: TBitmap;
begin
  if OutlookList.Selected = nil then Exit;
  ABitmap := TBitmap.Create;
  try
    ImageList1.GetBitmap(OutlookList.Selected.Index, ABitmap);
    fcImager1.Picture.Assign(ABitmap);
  finally
    ABitmap.Free;
  end;
end;
end.

File: 06 - Exam Analysis/Vcl/other/w2w/1StClass4000/VCL6/cdemos/imager/TilerUnit.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "TilerUnit.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "fcImager"
#pragma link "fcLabel"
#pragma link "fcOutlookList"
#pragma link "fcButton"
#pragma link "fcButtonGroup"
#pragma link "fcClearPanel"
#pragma link "fcImgBtn"
#pragma link "fcOutlookBar"
#pragma link "fcShapeBtn"
#pragma resource "*.dfm"
TTilerDemoForm *TilerDemoForm;
//---------------------------------------------------------------------------
__fastcall TTilerDemoForm::TTilerDemoForm(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TTilerDemoForm::fcControlBar1OutlookList1ItemChange(
      TfcCustomOutlookList *OutlookList, TfcOutlookListItem *Item)
{
  if (OutlookList->Selected == NULL) return;

  Graphics::TBitmap* ABitmap = new Graphics::TBitmap;
  ImageList1->GetBitmap(OutlookList->Selected->Index, ABitmap);
  fcImager1->Picture->Assign(ABitmap);
  ABitmap->Free();
}
//---------------------------------------------------------------------------
